src/Http/Controllers/VoyagerProfileController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace TCG\Voyager\Http\Controllers;
+
+use Illuminate\Http\Request;
+use TCG\Voyager\Events\BreadDataUpdated;
+use TCG\Voyager\Facades\Voyager;
+
+class VoyagerProfileController extends Controller
+{
+    public function index()
+    {
+        return Voyager::view('voyager::profile');
+    }
+
+    // POST BR(E)AD
+    public function update(Request $request)
+    {
+        $slug = 'users';
+
+        $dataType = Voyager::model('DataType')->where('slug', '=', $slug)->first();
+
+        $id = $request->user()->id;
+
+        $data = call_user_func([$dataType->model_name, 'findOrFail'], $id);
+
+        // Check permission
+        $this->authorize('edit', $data);
+
+        //Verify if user tries to modify his own role
+        if ($request->has('role_id')) {
+            $request->merge(['role_id' => $request->user()->role->id]);
+        }
+
+        if($request->has('user_belongstomany_role_relationship')){
+            $request->offsetUnset('user_belongstomany_role_relationship');
+        }
+
```

Comment:
I already reviewed this, but I kind of panicked when I thought I made a mistake and removed it without thinking....

I think you also need to remove the `user_belongsto_role_relationship` field from the request.

```php
if ($request->has('user_belongsto_role_relationship')) {
    $request->offsetUnset('user_belongsto_role_relationship');
}
```

I was able to change my default role by inserting the following input to the webpage: `<input type="text" class="form-control" name="user_belongsto_role_relationship" value="!! RANDOM_NUMBER !!">`

Reply:
Thanks for the reply.. I'll check here and submit another commit solving this problem!